sql/src/test/java/io/crate/integrationtests/DiskUsagesITest.java

```diff
@@ -92,9 +92,11 @@ public void testRerouteOccursOnDiskPassingHighWatermark() throws Exception {
         ensureGreen();
         {
             var shardCountByNodeId = getShardCountByNodeId();
-            assertThat("node0 has at least 3 shards", shardCountByNodeId.get(nodeIds.get(0)), greaterThanOrEqualTo(3));
-            assertThat("node1 has at least 3 shards", shardCountByNodeId.get(nodeIds.get(1)), greaterThanOrEqualTo(3));
-            assertThat("node2 has at least 3 shards", shardCountByNodeId.get(nodeIds.get(2)), greaterThanOrEqualTo(3));
+            assertBusy(() -> {
```

Comment:
this won't fix it such as `shardCountByNodeId` won't be updated in the busy loop. you have to add it on the line 93.